packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -302,6 +479,34 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
         store.payOracleFeesErc20(address(collateralCurrency), amount);
     }
 
+    function _calculateProposalRewardPct(
+        address perpetual,
+        uint256 startTime,
+        uint256 endTime,
+        FixedPoint.Signed memory proposedRate,
+        FixedPoint.Signed memory currentRate
+    ) private view returns (FixedPoint.Unsigned memory reward) {
+        uint256 timeDiff = endTime.sub(startTime);
+
+        FixedPoint.Unsigned memory rewardRate = _getFundingRateRecord(perpetual).rewardRatePerSecond;
+
+        // First compute the reward for the time elapsed.
+        reward = rewardRate.mul(timeDiff);
+
+        // Next scale the reward based on the absolute difference % between the current and proposed rates.
+        FixedPoint.Signed memory diff = (
+            currentRate.isGreaterThan(proposedRate) ? currentRate.sub(proposedRate) : proposedRate.sub(currentRate)
+        );
+        FixedPoint.Signed memory diffPercent = (currentRate.isEqual(0) ? diff : diff.div(currentRate));
+        FixedPoint.Unsigned memory absDiffPercent = (
+            diffPercent.isLessThan(FixedPoint.fromUnscaledInt(0))
+                ? FixedPoint.fromSigned(diffPercent.mul(FixedPoint.fromUnscaledInt(-1)))
+                : FixedPoint.fromSigned(diffPercent)
+        );
+        // TODO: Should we set ceiling on this
+        reward = reward.mul(absDiffPercent.add(1));
```

Comment:
I think we should either consider capping this or using a different functional form (maybe not in this PR) -- what's to stop a funding rate that's very close to 0 causing the _next_ reward to be so large that it bankrupts the contract?

Reply:
Agreed this is why I think it makes sense to set a cap, but I'm curious what you have in mind for other functions that could achieve the same effect of "scaling" the proposers reward proportional to the magnitude of the update

Reply:
For now I've set a 200% ceiling on the diff %